### DISCUSSION TITLE
Roadmap/Docs

### DISCUSSION THREAD

**Discussion by RyanPaulGannon** [toniengelhardt/merklin#30] [General]
Maybe I'm getting a bit ahead here I don't know. But is a Roadmap/Documentation something we should start looking in too?

**Comment by toniengelhardt** [comment]
You mean for users or developers?

**Reply by RyanPaulGannon** [reply]
First thought was for devs, but I suppose for users could be good too?

**Reply by toniengelhardt** [reply]
I think for devs we can have it on GitHub for now, or try Docus. Or even better maybe add it directly to the code as comments. Maybe a combination of both actually. What do you think?

**Reply by toniengelhardt** [reply]
Generally I think it should be very easy to join the project and get going fast without having to tinker around too much.

**Reply by toniengelhardt** [reply]
It's probably a good rule of thumb to try to have the documentation always as close as possible to the code that it concerns. That way it should be easiest to find.

**Reply by RyanPaulGannon** [reply]
I would agree too, I'd read about it in a book and whilst I've read docs etc, I've never been involved in writing/implementing 😅. 

I've found it fairly easy to pick up quickly too, so I'd fully expect other, better devs to easily navigate their way around!

**Reply by toniengelhardt** [reply]
But probably gets more difficult that more complex the project becomes. Therefore, clarity should probably be the main goal for the code. 

One experience I already had several times with dedicated docs is that it gets out of sync with the code aka. people write the docs when they first create the code, but then update the code and not the docs. The problem maybe being that when you change code it's hard to find all the docs that are related. That actually even happens with docs that that are in the code (comments, docstrings), even though this should be much easier to handle.

My personal opinion is that the comment/docstring should always be regarded as the source of truth and if the code does something different it is a bug.

**Reply by RyanPaulGannon** [reply]
Yeah, I like that idea. I'd forgotten about docstrings (mainly as I've not used them a great deal), but that should be very helpful!
